Jenkinsfile

```diff
@@ -1125,6 +1128,48 @@ def isDocChangedOnly(){
   }
 }
 
+/**
+  Often developers focus on changes to a specific module in a single PR. If that's the case, this function will return the
+  name of the changed module so we can skip tests for all other modules
+**/
+def moduleToTest() {
+  if (params.runAllStages || !env.CHANGE_ID?.trim()) {
+    log(level: 'INFO', text: 'Speed build for specific module only is disabled for branches/tags or when forcing with the runAllStages parameter.')
+    return 'false'
+  }
+
+  def modulePattern = /beat\/module\/([^\/]+)/
+  def module = ""
+  filesChanged().each{ String file ->
+    matches = file =~ modulePattern
+    if (matches.size == 2) {
+      matchedModule = matches[1]
+      if (module == "") {
+        // Module not initialized; initialize it with matched module.
+        module = matchedModule
+      } else if (module != matchedModule) {
+        // Module already initialized and is different from matched module. Multiple
+        // modules were changed, so don't return any specific module.
+        return ""
+      }
+    }
+  }
+
+  return module
+}
+
+/**
+ TODO: move into https://github.com/elastic/apm-pipeline-library/ and potentially reuse in isGitRegionMatch definition.
+ TODO: add error checking for from/to.
+**/
+def filesChanged(Map params = [:]) {
+  def from = params.get('from', env.CHANGE_TARGET?.trim() ? "origin/${env.CHANGE_TARGET}" : env.GIT_PREVIOUS_COMMIT)
```

Comment:
It would only evaluate the changes in the latest commit, it can be fine meanwhile you are making changes but it is not when you have to test all the changes in the PR. Also, we make a merge with the target branch before test the PR so can exist changes that are not in the PR and it is expected.